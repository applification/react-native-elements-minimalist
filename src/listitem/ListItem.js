import React, { Component } from 'react';
import { View, Text } from 'react-primitives';
import PropTypes from 'prop-types';

class ListItemComponent extends Component {
  static propTypes = {
    title: PropTypes.string,
  };

  render() {
    const { title } = this.props;
    return (
      <View
        style={{
          backgroundColor: '#F6F6F6',
          paddingLeft: 20,
          paddingTop: 20,
          paddingBottom: 20,
        }}
      >
        <Text>{title}</Text>
      </View>
    );
  }
}

export default ListItemComponent;
