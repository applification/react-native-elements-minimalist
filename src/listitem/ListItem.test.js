import React from 'react';
import renderer from 'react-test-renderer';

import ListItem from './ListItem';

describe('LISTITEM', () => {
  const title = 'List item title';

  it('Renders a list item', () => {
    const component = renderer.create(<ListItem title={title} />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
