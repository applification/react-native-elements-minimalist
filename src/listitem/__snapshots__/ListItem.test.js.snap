// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LISTITEM Renders a list item 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F6F6F6",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingTop": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    List item title
  </Text>
</View>
`;
